### ISSUE TITLE
Better support for removing unused open statements

### ISSUE THREAD

**Issue by BentTranberg** [dotnet/fsharp#8682]
I have just gone through half an hour of removing unused open statements from my big solution. I am not the kind of developer that is sloppy and leaves behind untidy code, and yet the situation with Visual Studio and F# development is such that I can't afford continuously checking for unused open statements with the available tooling. Because of this, unused open statements will accumulate over time. I have to do cleanups regularly at intervals of weeks or months, which is not optimal. This situation should be rectified with improved tooling.

The current situation in Visual Studio is that unused open statements are greyed out. This is not sufficiently efficient for me.

* I can't afford the time it takes to check all the places in the file where open statements can be placed. Unlike C#, open statements can be placed pretty much anywhere in all modules in files. Also, the time the analyzer spends greying out, can be too slow. I've usually gone elsewhere before the problem becomes visible after I'm done editing. Don't ask me to hold on for a second.

* The greying out is not sufficiently distinguishable from normal text on quite a few monitors. This is an issue in itself, which actually prevents me checking for open statements at all on some of my computers, and contributes greatly to the accumulation of unused open statements.

Long ago, when working with Java development for a few months, I enjoyed an editor that automatically inserted and even removed the Java equivalent of open statements needed for the code written, while editing. I don't necessarily need it to be that advanced, but if it is then I'd like to have some control of the order and placement myself.

I don't want the job relegated to an external tool or even an internal analyzer. That would mean I would have to do a tedious job, however small, in addition to my fast innermost development cycle, and it would still not be affordable on every edit.

Rather I want this to be handled in the same way as any other incorrect source - in other words I want a warning for unused open statements. That would be simple to implement, and effective. Making this warning an error in the project can help guarantee there won't be unused opens.

Analysis for unused opens can fail in rare cases (there are issues in here because of that), so there must be some way to turn off the warning for a single open statement. It would also be smart to allow a way for certain namespaces, e.g. System, to be opened without giving a warning.

**Comment by BentTranberg** [comment]
The situation has improved considerably for me since I first reported, partly due to improvements in Visual Studio and F# tooling, and partly due to my new hardware.

* The greying out of unused open statements is now updated so fast that I can easily remove unused open statements while I work on the related source.
* The colors of "used" and "unused" are now very distinct in Dark as well as Light themes, so I am very well aware of any unused opens while they are visible on the screen.

I have changed hardware all over the place since I first reported, so I can't tell for sure which causes the improvements and by how much, but I guess that in any case there have been improvements in F# tooling and the color schemes.

This means I now use this feature near 100% the way I want.

What I could still wish for, is a warning that an open statement is unused. That would plug the final hole preventing me from keeping the source completely clean of unused opens.
